src/utils/UWebSocketTransport.ts

```diff
@@ -35,80 +43,159 @@ export async function createUWebSocketTransport<T extends { isConnected: boolean
+  const retryDelayMs = (await config.getNumber('WS_TRANSPORT_RETRY_DELAY_MS')) || 1000
+  const maxRetryAttempts = (await config.getNumber('WS_TRANSPORT_MAX_RETRY_ATTEMPTS')) || 5
+  const maxBackoffDelayMs = (await config.getNumber('WS_TRANSPORT_MAX_BACKOFF_DELAY_MS')) || 30000 // 30 seconds maximum delay
 
   let isTransportActive = true
   let isInitialized = false
+
+  // Simple queue for messages that couldn't be sent immediately
+  const messageQueue: Array<{
+    message: Uint8Array
+    future: IFuture<void>
+    attempts: number
+  }> = []
+
   let isProcessing = false
-  let queueProcessingTimeout: NodeJS.Timeout | null = null
+  let processingTimeout: NodeJS.Timeout | null = null
+
+  type QueuedMessage = {
+    message: Uint8Array
+    future: IFuture<void>
+    attempts: number
+  }
 
   async function processQueue() {
     if (isProcessing || !isTransportActive || !isInitialized) {
-      logger.debug('[DEBUGGING CONNECTION] Queue processing skipped', {
-        transportId,
-        isProcessing: String(isProcessing),
-        isTransportActive: String(isTransportActive),
-        isInitialized: String(isInitialized),
-        queueLength: messageQueue.length
-      })
       return
     }
 
-    logger.debug('[DEBUGGING CONNECTION] Processing queue', {
-      transportId,
-      queueLength: messageQueue.length
-    })
-
     isProcessing = true
-
-    if (queueProcessingTimeout) {
-      clearTimeout(queueProcessingTimeout)
-      queueProcessingTimeout = null
+    if (processingTimeout) {
+      clearTimeout(processingTimeout)
+      processingTimeout = null
     }
 
     try {
       while (messageQueue.length > 0 && isTransportActive && socket.getUserData().isConnected) {
-        const item = messageQueue[0]
-        try {
-          const result = socket.send(item.message, true)
-          if (result === 0) {
-            // Send failed, socket might be closed
-            item.future.reject(new Error('Failed to send message: Socket closed'))
-            messageQueue.shift()
-            cleanup()
-            return
-          }
-          item.future.resolve()
+        const currentMessage = messageQueue[0]
+
+        // Check max retries
+        if (currentMessage.attempts >= maxRetryAttempts) {
+          logger.warn('Message dropped after max retries', {
+            transportId,
+            attempts: currentMessage.attempts,
+            messageSize: currentMessage.message.byteLength
+          })
+          currentMessage.future.reject(new Error('Message dropped after max retries'))
           messageQueue.shift()
-        } catch (error: any) {
-          item.future.reject(new Error(`Failed to send message: ${error.message}`))
-          messageQueue.shift()
-          cleanup()
-          return
+          continue
+        }
+
+        const result = processNextMessage(currentMessage)
+
+        if (result === UWebSocketSendResult.SUCCESS || result === UWebSocketSendResult.ERROR) {
```

Comment:
Whenever the [backpressure status code is returned](https://github.com/uNetworking/uWebSockets/blob/master/misc/READMORE.md#backpressure), it means that the message fits in the backpressure buffer and will be queued to be sent whenever it's possible. So that means that we shouldn't retry sending the message to the user.
We should only retry it when there's a DROPPED result, which means that the backpressure is too high to be a able to send a message.
```suggestion
        if (result === UWebSocketSendResult.SUCCESS || result === UWebSocketSendResult.ERROR || result === UWebSocketSendResult.ERROR) {
```